openwisp-config/files/openwisp.init

```diff
@@ -88,7 +89,7 @@ start_service() {
 	                        $cacert $mac_interface $management_interface $default_hostname \
 	                        $pre_reload_hook $post_reload_hook $post_reload_delay \
 	                        $post_registration_hook
-	procd_set_param respawn
+	procd_set_param respawn 3600 ${respawn_timeout:-120} 0
```

Comment:
Am I interpreting correclty that setting the last parameter to zero means retrying indefinitely?
Over time I have observed that if it crashes repeatedly there's a misconfiguration and retrying indefinitely will not help but may fill the logs and make it harder to debug.
So what is the cause that is urging us to retry indefinitely? Is there another issue to handle somwhere else?

Reply:
> Am I interpreting correclty that setting the last parameter to zero means retrying indefinitely?

Yes, that's correct.

> Over time I have observed that if it crashes repeatedly there's a misconfiguration and retrying indefinitely will not help but may fill the logs and make it harder to debug.
> So what is the cause that is urging us to retry indefinitely? Is there another issue to handle somwhere else?

In our opinion, the openwisp agent is a service of such high importance that it should be kept alive at all costs.

For example, a server problem or misconfiguration on the server side should never cause the CPEs to stop trying to contact the server.

But if e.g. the shared_secret on the server is wrong by mistake, this may cause all CPEs to give up trying to contact the server.